### PR TITLE
Extend evaluate syntax

### PR DESCRIPTION
**Pull request by ianspektor**
- [X] allow passing data keyed by event name to `evaluate` (e.g. `tp.evaluate(X, {"my_event": evset})`)
    - [X] write basic test
- [x] allow calling `.evaluate` on a node and use it as query
    - [x] write basic test
- [x] allow passing list of events and single event as input data to `evaluate` and use their name to find the corresponding node (e.g. `tp.evaluate(X, [evset_1, evset_2]))`
    - [x] write basic test
- [x] write comprehensive tests

### CODE REVIEW

File: temporian/core/test/evaluation_test.py [hunk 1]
```diff
@@ -113,6 +113,76 @@ def test_evaluate_dict(self):
...
+        e3 = utils.create_input_event_set("i3")
+        i3 = e3.node()
+
+        result = evaluation.evaluate([i3, i1, i2], [e1, e2, e3])
+
+        self.assertIsInstance(result, list)
+        self.assertTrue(result[0] is e3)
+        self.assertTrue(result[1] is e1)
+        self.assertTrue(result[2] is e2)
+
```
Comment (DonBraulio) [inline temporian/core/test/evaluation_test.py:185]: I'd add a test to check failure on duplicate node names

File: temporian/core/evaluation.py [hunk 2]
```diff
@@ -16,34 +16,50 @@
...
-from temporian.implementation.numpy import evaluator as numpy_evaluator
+from temporian.implementation.numpy import evaluation as np_eval
+from temporian.implementation.numpy.data.event_set import EventSet
+
 
-AvailableBackends = Any
-Data = Dict[Node, Union[str, pathlib.Path, EventSet]]
-Query = Union[Node, List[Node], Dict[str, Node]]
-Result = Union[EventSet, List[EventSet], Dict[str, EventSet]]
+EvaluationQuery = Union[Node, List[Node], Dict[str, Node]]
```
[the remark this answers is not in the document]
Comment (ianspektor) [reply temporian/core/evaluation.py:29]: its used to return a dict with same keys as the input, if the input is a dict. I agree its unclear what they should contain - because its arbitrary, can be anything. will add that to the docstring.

```
If `query` is a dictionary of nodes, the returned object will be a
        dictionary of event sets.
```

File: temporian/core/processor.py [hunk 3]
```diff
@@ -145,22 +118,26 @@ def p2(title, dictionary):
 
 
 def infer_processor(
```
[the remark this answers is not in the document]
Comment (ianspektor) [reply temporian/core/processor.py:120]: we aren't checking, I'm guessing it would just take the first one it finds. I'll add a test and adjust accordingly 👍🏼

File: temporian/core/test/evaluation_test.py (hunk 1 above)
Comment (ianspektor) [reply temporian/core/test/evaluation_test.py:185]: added three

**Review by DonBraulio — APPROVED** [review]
Niceee! 🙌
